include/roboteam_utils/Optimization.h

```diff
@@ -69,7 +69,7 @@ namespace rtt {
          * the same upper and lower limits.
          */
         GradientDescent(ScoreFunction scorer, Data initialData, Num initialStep, Num minimum, Num maximum)
-                : scorer(scorer), data(data), lastScore(-99999), justFlipped(true), iterationCount(0) {
+                : scorer(scorer), data(initialData), lastScore(-99999), justFlipped(true), iterationCount(0) {
```

Comment:
what does this change mean?

Reply:
Previously it was self-assigning the field (data was initialized with data), this is a bug. This seems like the most sensible fix.